### PR TITLE
Add deprecation logging for lenient boolean queries

### PR DESCRIPTION
**Pull request by dakrone**
Relates to #21555, this is the deprecation logging for what was removed.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/index/mapper/BooleanFieldMapper.java [hunk 1]
```diff
@@ -53,6 +55,8 @@
 
     public static final String CONTENT_TYPE = "boolean";
 
+    private static DeprecationLogger deprecationLogger = new DeprecationLogger(Loggers.getLogger(BooleanFieldMapper.class));
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/index/mapper/BooleanFieldMapper.java]: final?

File: docs/reference/mapping/types/boolean.asciidoc [hunk 2]
```diff
@@ -13,6 +13,10 @@ True values::
 
     Anything that isn't false.
 
+deprecated[5.1.0 While Elasticsearch will currently accept the above values during index time.
+Searching a boolean field using these psuedo-boolean values is deprecated.
```
Comment (jpountz) [inline docs/reference/mapping/types/boolean.asciidoc]: pseudo

**Review by jpountz — APPROVED** [review]
LGTM

File: core/src/main/java/org/elasticsearch/index/mapper/BooleanFieldMapper.java (hunk 1 above)
Comment (dakrone) [inline core/src/main/java/org/elasticsearch/index/mapper/BooleanFieldMapper.java]: Definitely, will change.

File: docs/reference/mapping/types/boolean.asciidoc (hunk 2 above)
Comment (dakrone) [inline docs/reference/mapping/types/boolean.asciidoc]: Whoops thanks :)

**Review by jpountz — APPROVED** [review]
LGTM

**Pull request merged**